src/zenml/integrations/slack/flavors/slack_alerter_flavor.py

```diff
@@ -15,10 +15,16 @@
 
 from typing import TYPE_CHECKING, Optional, Type
 
+from slack_sdk import WebClient
```

Comment:
Actually I just realised another issue: ZenML imports this flavor file (and calls `StackComponentConfig.is_valid`) when the slack alerter is registered. However, this can also happen server side or you can even do it locally without having the slack integration installed. In those cases, it would now crash because the `slack_sdk` library is not installed. To avoid this, I suggest moving the import into the `is_valid` property and handling the case when it's not installed:
```python
@property
def is_valid(self) -> bool:
  try:
    from slack_sdk import WebClient
    from slack_sdk.errors import SlackApiError
  except ImportError:
    logger.warning("Unable to validate Slack alerter credentials because the Slack integration is not installed.")
    return
  
  ...  # do the actual validation here
```